ngraph/src/ngraph/runtime/reference/slice.hpp

```diff
@@ -19,40 +19,26 @@
 #include <cmath>
 
 #include "ngraph/check.hpp"
+
 #include "ngraph/coordinate_transform.hpp"
+#include "ngraph/type/element_type.hpp"
 
 namespace ngraph
 {
     namespace runtime
     {
         namespace reference
         {
-            template <typename T>
-            void slice(const T* arg,
-                       T* out,
-                       const Shape& arg_shape,
-                       const Coordinate& lower_bounds,
-                       const Coordinate& upper_bounds,
-                       const Strides& strides,
-                       const Shape& out_shape)
-            {
-                CoordinateTransform input_transform(arg_shape, lower_bounds, upper_bounds, strides);
-                CoordinateTransform output_transform(out_shape);
-
-                CoordinateTransform::Iterator output_it = output_transform.begin();
-
-                NGRAPH_CHECK(shape_size(input_transform.get_target_shape()) ==
-                             shape_size(output_transform.get_target_shape()));
-
-                for (const Coordinate& in_coord : input_transform)
-                {
-                    const Coordinate& out_coord = *output_it;
-
-                    out[output_transform.index(out_coord)] = arg[input_transform.index(in_coord)];
-
-                    ++output_it;
-                }
-            }
+            // NGRAPH_API should be removed here
+            // slice is exposed as public only for test INTERPRETER backend
+            NGRAPH_API void slice(const char* arg,
```

Comment:
@ilyachur Any idea how to avoid expose it to public API?
I've considered:
- indicate it as `inline` and move implementation to header, but it can increase binary size
- keep old template slice only for Interpreter, but I am not sure if maintaining two versions is good idea

Reply:
Can Interpreter use evaluate method instead of using `reference::slice` implementation?

As I know @mikhail-treskin does something like that.

Reply:
Good point, I added evaluate method to legacy slice op.